src/generic_ui/scripts/user.ts

```diff
@@ -41,46 +40,61 @@ module UI {
       }
       this.name = profile.name;
       this.imageData = profile.imageData || UI.DEFAULT_USER_IMG;
-      this.isOnline = profile.isOnline;
     }
 
     // Returns two strings, where each matches an array name in model.contacts.
     // Does not use an enum because we need a string value, and typescript
     // enums evaluate to numbers.
```

Comment:
I still think getCategories should return a string.  Maybe that string can be Category[Category.onlineTrustedUproxy], but I don't think we should require that the callers of this method all need to do typescript enum conversion.

Typescript enums have not saved me time personally, I've spent a while debugging this and the issue with clientState.

Also regarding Will's comment - this isn't just an issue with console messages (where we can't print an enum value but have to convert it), but also an issue when we are looking in the Chrome JavaScript debugger - if an enum value is passed around there it will appear a number and not a string.